server/routerlicious/packages/routerlicious-base/src/riddler/tenantManager.ts

```diff
@@ -92,19 +96,20 @@ export class TenantManager {
                 }
             }
         }
-        
         // If Key 1 validation fails, try with Key 2
         try {
             await this.validateTokenWithKey(tenantKeys.key2, KeyName.key2, token);
         } catch (error) {
             if (isNetworkError(error)) {
                 if (error.code === 403) {
                     // Delete key from cache on 403
-                    Lumberjack.info(`Error with key 2 token validation. Deleting key from cache.`, lumberProperties);
-                    await this.deleteKeyFromCache(tenantId).catch((err) => {
-                        Lumberjack.error(`Error deleting keys from the cache.`, lumberProperties, err);
-                    });
-                    if (bypassCache === false) {
+                    if (bypassCache && this.isCacheEnabled) {
+                        Lumberjack.info(`Error with key 2 token validation. Deleting key from cache.`, lumberProperties);
+                        await this.deleteKeyFromCache(tenantId).catch((err) => {
+                            Lumberjack.error(`Error deleting keys from the cache.`, lumberProperties, err);
+                        });
+                    }
+                    if (bypassCache === false && this.isCacheEnabled) {
```

Comment:
You can do `!bypassCache` instead of `bypassCache === false`. Or, if we adopt the same naming convention everywhere: `!disableCache`

Reply:
Changed to !bypassCache because I thought this was a more accurate variable name